### DISCUSSION TITLE
Valet share error:Tunnel 'command_line' specifies an invalid or malformed address

### DISCUSSION THREAD

**Discussion by spawn81** [laravel/valet#1148] [Q&A]
## Clear description of your problem

I have a Mac M1 and I want to share a site using valet share,  but I have this error:
 **Tunnel 'command_line' specifies an invalid or malformed address 'tailtest.**
The tld is tailtest.test, it's served over https and locally it's working.
I read the #790 so I tried 
**ngrok http ${PWD##*/}.test:80 -host-header=rewrite**
 but got 503 error after clicking on the links.
## Expected behavior

getting the ngrok shareable links
## Current behavior

 **Tunnel 'command_line' specifies an invalid or malformed address 'tailtest.** or 503 error
## Steps to Reproduce

1. valet share

### Output of these steps

NAME:
   http - start an HTTP tunnel

USAGE:
   ngrok http [command options] [arguments...]

DESCRIPTION:
   Starts a tunnel listening for HTTP/HTTPS traffic with a specific hostname.
   The HTTP Host header on incoming public requests is inspected to
   determine which tunnel it matches.

   HTTPS endpoints terminate TLS traffic at the ngrok server using the
   ngrok.io certificates. The decrypted, HTTP traffic is then forwarded
   through the secure tunnel and then to your local server. If you don't want
   your TLS traffic to terminate at the ngrok server, use a TLS or TCP tunnel.

EXAMPLES:
   ngrok http 8080                      # forward ngrok.io subdomain to port 80
   ngrok http example.com:9000          # forward traffic to example.com:9000
   ngrok http -subdomain=bar 80         # request subdomain name: 'bar.ngrok.io'
   ngrok http -hostname=ex.com 1234     # request tunnel 'ex.com' (DNS CNAME)
   ngrok http -auth='falken:joshua' 80  # enforce basic auth on tunnel endpoint
   ngrok http -host-header=ex.com 80    # rewrite the Host header to 'ex.com'
   ngrok http file:///var/log           # serve local files in /var/log
   ngrok http https://localhost:8443    # forward to a local https server

OPTIONS:
   --auth 		enforce basic auth on tunnel endpoint, 'user:password'
   --authtoken 		ngrok.com authtoken identifying a user
   --bind-tls "both"	listen for http, https or both: true/false/both
   --config		path to config files; they are merged if multiple
   --host-header 	set Host header; if 'rewrite' use local address hostname
   --hostname 		host tunnel on custom hostname (requires DNS CNAME)
   --inspect		enable/disable http introspection
   --log "false"	path to log file, 'stdout', 'stderr' or 'false'
   --log-format "term"	log record format: 'term', 'logfmt', 'json'
   --log-level "info"	logging level
   --region 		ngrok server region [us, eu, au, ap, sa, jp, in] (default: us)
   --subdomain 		host tunnel on a custom subdomain

ERROR:  Tunnel 'command_line' specifies an invalid or malformed address 'tailtest.
ERROR:  Deprecated: Return type of Tightenco\Collect\Support\Collection::offsetExists($key) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1411
ERROR:  
ERROR:  Deprecated: Return type of Tightenco\Collect\Support\Collection::offsetGet($key) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1422
ERROR:  
ERROR:  Deprecated: Return type of Tightenco\Collect\Support\Collection::offsetSet($key, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1434
ERROR:  
ERROR:  Deprecated: Return type of Tightenco\Collect\Support\Collection::offsetUnset($key) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1449
ERROR:  
ERROR:  Deprecated: Return type of Tightenco\Collect\Support\Collection::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1366
ERROR:  
ERROR:  Deprecated: Return type of Tightenco\Collect\Support\Collection::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1356
ERROR:  
ERROR:  Deprecated: Return type of Tightenco\Collect\Support\Collection::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/Traits/EnumeratesValues.php on line 842
ERROR:  
ERROR:  Deprecated: Return type of Tightenco\Collect\Support\LazyCollection::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/LazyCollection.php on line 1350
ERROR:  
ERROR:  Deprecated: Return type of Tightenco\Collect\Support\LazyCollection::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/LazyCollection.php on line 1340
ERROR:  
ERROR:  Deprecated: Return type of Tightenco\Collect\Support\LazyCollection::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/Traits/EnumeratesValues.php on line 842
ERROR:  test:60': strange hostname
## Diagnosis
<details>
<summary>sw_vers</summary>
<pre>ProductName:	macOS
ProductVersion:	11.6
BuildVersion:	20G165</pre>
</details>
<details>
<summary>valet --version</summary>
<pre>Deprecated: Return type of Tightenco\Collect\Support\Collection::offsetExists($key) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1411

Deprecated: Return type of Tightenco\Collect\Support\Collection::offsetGet($key) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1422

Deprecated: Return type of Tightenco\Collect\Support\Collection::offsetSet($key, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1434

Deprecated: Return type of Tightenco\Collect\Support\Collection::offsetUnset($key) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1449

Deprecated: Return type of Tightenco\Collect\Support\Collection::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1366

Deprecated: Return type of Tightenco\Collect\Support\Collection::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/Collection.php on line 1356

Deprecated: Return type of Tightenco\Collect\Support\Collection::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/Traits/EnumeratesValues.php on line 842

Deprecated: Return type of Tightenco\Collect\Support\LazyCollection::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/LazyCollection.php on line 1350

Deprecated: Return type of Tightenco\Collect\Support\LazyCollection::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/LazyCollection.php on line 1340

Deprecated: Return type of Tightenco\Collect\Support\LazyCollection::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/leon/.composer/vendor/tightenco/collect/src/Collect/Support/Traits/EnumeratesValues.php on line 842
Laravel Valet 2.16.1</pre>
</details>
<details>
<summary>cat ~/.config/valet/config.json</summary>
<pre>{
    "tld": "test",
    "loopback": "127.0.0.1",
    "paths": [
        "/Users/leon/Documents/Development/web"
    ]
}</pre>
</details>
<details>
<summary>cat ~/.composer/composer.json</summary>
<pre>{
    "require": {
        "laravel/installer": "^4.2",
        "laravel/valet": "^2.16"
    }
}</pre>
</details>
<details>
<summary>composer global diagnose</summary>
<pre>Changed current directory to /Users/leon/.composer
Checking composer.json: WARNING
No license specified, it is recommended to do so. For closed-source software you may use "proprietary" as license.
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys: FAIL
Missing pubkey for tags verification
Missing pubkey for dev verification
Run composer self-update --update-keys to set them up
Checking composer version: OK
Composer version: 2.1.12
PHP version: 8.1.0
PHP binary path: /opt/homebrew/Cellar/php/8.1.0/bin/php
OpenSSL version: OpenSSL 1.1.1l  24 Aug 2021
cURL version: 7.80.0 libz 1.2.11 ssl (SecureTransport) OpenSSL/1.1.1l
zip: extension present, unzip present, 7-Zip not available</pre>
</details>
<details>
<summary>composer global outdated</summary>
<pre>Changed current directory to /Users/leon/.composer
Legend:
! patch or minor release available - update recommended
~ major release available - update possible
illuminate/container          v8.69.0 ! v8.73.2 The Illuminate Container package.
illuminate/contracts          v8.69.0 ! v8.73.2 The Illuminate Contracts package.
psr/container                 1.1.2   ~ 2.0.2   Common Container Interface (PHP FIG PSR-11)
psr/simple-cache              1.0.1   ~ 3.0.0   Common interfaces for simple caching
symfony/console               v5.3.10 ~ v6.0.0  Eases the creation of beautiful and testable command line interfaces
symfony/deprecation-contracts v2.4.0  ~ v3.0.0  A generic function and convention to trigger deprecation notices
symfony/process               v5.3.7  ~ v6.0.0  Executes commands in sub-processes
symfony/service-contracts     v2.4.0  ~ v3.0.0  Generic abstractions related to writing services
symfony/string                v5.3.10 ~ v6.0.0  Provides an object-oriented API to strings and deals with bytes, UTF-8 code points and grapheme clusters in a unified way
symfony/var-dumper            v5.3.10 ~ v6.0.0  Provides mechanisms for walking through any arbitrary PHP variable</pre>
</details>
<details>
<summary>ls -al /etc/sudoers.d/</summary>
<pre>total 0
drwxr-xr-x   2 root  wheel    64 10 Set 02:32 .
drwxr-xr-x  86 root  wheel  2752 29 Nov 23:41 ..</pre>
</details>
<details>
<summary>brew config</summary>
<pre>HOMEBREW_VERSION: 3.3.6
ORIGIN: https://github.com/Homebrew/brew
HEAD: 32a3e1fe28f9372a6c6d5eca52129f92b0f00ba0
Last commit: 33 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: c4966e37d45f2b13bffa46f7575041b8c344d86f
Core tap last commit: 64 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 13.0.0 build 1300
Git: 2.30.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.6-arm64
CLT: 13.0.0.0.1.1630607135
Xcode: N/A
Rosetta 2: false</pre>
</details>
<details>
<summary>brew services list</summary>
<pre>Name    Status  User File
dnsmasq stopped      
httpd   stopped      
mysql   started leon 
nginx   stopped      
php     started root</pre>
</details>
<details>
<summary>brew list --formula --versions | grep -E "(php|nginx|dnsmasq|mariadb|mysql|mailhog|openssl)(@\d\..*)?\s"</summary>
<pre>dnsmasq 2.86
mysql 8.0.27
nginx 1.21.4
openssl@1.1 1.1.1l_1
php 8.1.0</pre>
</details>
<details>
<summary>brew outdated</summary>
<pre>python@3.9</pre>
</details>
<details>
<summary>brew tap</summary>
<pre>homebrew/cask
homebrew/core
homebrew/services
shivammathur/php</pre>
</details>
<details>
<summary>php -v</summary>
<pre>PHP 8.1.0 (cli) (built: Nov 29 2021 23:05:16) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.0, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.0, Copyright (c), by Zend Technologies</pre>
</details>
<details>
<summary>which -a php</summary>
<pre>/opt/homebrew/bin/php
/usr/bin/php</pre>
</details>
<details>
<summary>php --ini</summary>
<pre>Configuration File (php.ini) Path: /opt/homebrew/etc/php/8.1
Loaded Configuration File:         /opt/homebrew/etc/php/8.1/php.ini
Scan for additional .ini files in: /opt/homebrew/etc/php/8.1/conf.d
Additional .ini files parsed:      /opt/homebrew/etc/php/8.1/conf.d/error_log.ini,
/opt/homebrew/etc/php/8.1/conf.d/ext-opcache.ini,
/opt/homebrew/etc/php/8.1/conf.d/php-memory-limits.ini</pre>
</details>
<details>
<summary>nginx -v</summary>
<pre>nginx version: nginx/1.21.4</pre>
</details>
<details>
<summary>curl --version</summary>
<pre>curl 7.64.1 (x86_64-apple-darwin20.0) libcurl/7.64.1 (SecureTransport) LibreSSL/2.8.3 zlib/1.2.11 nghttp2/1.41.0
Release-Date: 2019-03-27
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS GSS-API HTTP2 HTTPS-proxy IPv6 Kerberos Largefile libz MultiSSL NTLM NTLM_WB SPNEGO SSL UnixSockets</pre>
</details>
<details>
<summary>php --ri curl</summary>
<pre>curl

cURL support => enabled
cURL Information => 7.80.0
Age => 9
Features
AsynchDNS => Yes
CharConv => No
Debug => No
GSS-Negotiate => No
IDN => Yes
IPv6 => Yes
krb4 => No
Largefile => Yes
libz => Yes
NTLM => Yes
NTLMWB => Yes
SPNEGO => Yes
SSL => Yes
SSPI => No
TLS-SRP => Yes
HTTP2 => Yes
GSSAPI => Yes
KERBEROS5 => Yes
UNIX_SOCKETS => Yes
PSL => No
HTTPS_PROXY => Yes
MULTI_SSL => Yes
BROTLI => Yes
Protocols => dict, file, ftp, ftps, gopher, gophers, http, https, imap, imaps, ldap, ldaps, mqtt, pop3, pop3s, rtmp, rtsp, scp, sftp, smb, smbs, smtp, smtps, telnet, tftp
Host => arm-apple-darwin20.6.0
SSL Version => (SecureTransport) OpenSSL/1.1.1l
ZLib Version => 1.2.11
libSSH Version => libssh2/1.10.0

Directive => Local Value => Master Value
curl.cainfo => no value => no value</pre>
</details>
<details>
<summary>~/.composer/vendor/laravel/valet/bin/ngrok version</summary>
<pre>ngrok version 2.3.40</pre>
</details>
<details>
<summary>ls -al ~/.ngrok2</summary>
<pre>total 8
drwx------   3 leon  staff   96 30 Nov 00:57 .
drwxr-xr-x+ 23 leon  staff  736 30 Nov 01:09 ..
-rw-------   1 leon  staff   61 30 Nov 00:57 ngrok.yml</pre>
</details>
<details>
<summary>brew info nginx</summary>
<pre>nginx: stable 1.21.4 (bottled), HEAD
HTTP(S) server and reverse proxy, and IMAP/POP3 proxy server
https://nginx.org/
/opt/homebrew/Cellar/nginx/1.21.4 (26 files, 2.2MB) *
  Poured from bottle on 2021-11-08 at 00:45:03
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/nginx.rb
License: BSD-2-Clause
==> Dependencies
Required: openssl@1.1, pcre
==> Options
--HEAD
	Install HEAD version
==> Caveats
Docroot is: /opt/homebrew/var/www

The default port has been set in /opt/homebrew/etc/nginx/nginx.conf to 8080 so that
nginx can run without sudo.

nginx will load all files in /opt/homebrew/etc/nginx/servers/.

To restart nginx after an upgrade:
  brew services restart nginx
Or, if you don't want/need a background service you can just run:
  /opt/homebrew/opt/nginx/bin/nginx -g daemon off;
==> Analytics
install: 48,847 (30 days), 137,176 (90 days), 511,666 (365 days)
install-on-request: 48,748 (30 days), 136,854 (90 days), 510,456 (365 days)
build-error: 61 (30 days)</pre>
</details>
<details>
<summary>brew info php</summary>
<pre>php: stable 8.1.0 (bottled), HEAD
General-purpose scripting language
https://www.php.net/
/opt/homebrew/Cellar/php/8.1.0 (507 files, 81.6MB) *
  Built from source on 2021-11-30 at 00:05:56
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/php.rb
License: PHP-3.01
==> Dependencies
Build: httpd, pkg-config
Required: apr, apr-util, argon2, aspell, autoconf, curl, freetds, gd, gettext, gmp, icu4c, krb5, libpq, libsodium, libzip, oniguruma, openldap, openssl@1.1, pcre2, sqlite, tidy-html5, unixodbc
==> Options
--HEAD
	Install HEAD version
==> Caveats
To enable PHP in Apache add the following to httpd.conf and restart Apache:
    LoadModule php_module /opt/homebrew/opt/php/lib/httpd/modules/libphp.so

    <FilesMatch \.php$>
        SetHandler application/x-httpd-php
    </FilesMatch>

Finally, check DirectoryIndex includes index.php
    DirectoryIndex index.php index.html

The php.ini and php-fpm.ini file can be found in:
    /opt/homebrew/etc/php/8.1/

To restart php after an upgrade:
  brew services restart php
Or, if you don't want/need a background service you can just run:
  /opt/homebrew/opt/php/sbin/php-fpm --nodaemonize
==> Analytics
install: 103,355 (30 days), 223,526 (90 days), 684,971 (365 days)
install-on-request: 80,472 (30 days), 197,327 (90 days), 650,140 (365 days)
build-error: 89 (30 days)</pre>
</details>
<details>
<summary>brew info openssl</summary>
<pre>openssl@3: stable 3.0.0 (bottled) [keg-only]
Cryptography and SSL/TLS Toolkit
https://openssl.org/
Not installed
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/openssl@3.rb
License: Apache-2.0
==> Dependencies
Required: ca-certificates
==> Caveats
A CA file has been bootstrapped using certificates from the system
keychain. To add additional certificates, place .pem files in
  /opt/homebrew/etc/openssl@3/certs

and run
  /opt/homebrew/opt/openssl@3/bin/c_rehash

openssl@3 is keg-only, which means it was not symlinked into /opt/homebrew,
because macOS provides LibreSSL.

==> Analytics
install: 81,034 (30 days), 195,328 (90 days), 195,329 (365 days)
install-on-request: 68,490 (30 days), 160,635 (90 days), 160,636 (365 days)
build-error: 4,422 (30 days)</pre>
</details>
<details>
<summary>openssl version -a</summary>
<pre>LibreSSL 2.8.3
built on: date not available
platform: information not available
options:  bn(64,64) rc4(ptr,int) des(idx,cisc,16,int) blowfish(idx) 
compiler: information not available
OPENSSLDIR: "/private/etc/ssl"</pre>
</details>
<details>
<summary>openssl ciphers</summary>
<pre>ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-CHACHA20-POLY1305:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:GOST2012256-GOST89-GOST89:DHE-RSA-CAMELLIA256-SHA256:DHE-RSA-CAMELLIA256-SHA:GOST2001-GOST89-GOST89:AES256-GCM-SHA384:AES256-SHA256:AES256-SHA:CAMELLIA256-SHA256:CAMELLIA256-SHA:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-CAMELLIA128-SHA256:DHE-RSA-CAMELLIA128-SHA:AES128-GCM-SHA256:AES128-SHA256:AES128-SHA:CAMELLIA128-SHA256:CAMELLIA128-SHA:ECDHE-RSA-RC4-SHA:ECDHE-ECDSA-RC4-SHA:RC4-SHA:RC4-MD5:ECDHE-RSA-DES-CBC3-SHA:ECDHE-ECDSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:DES-CBC3-SHA</pre>
</details>
<details>
<summary>sudo nginx -t</summary>
<pre>nginx: the configuration file /opt/homebrew/etc/nginx/nginx.conf syntax is ok
nginx: configuration file /opt/homebrew/etc/nginx/nginx.conf test is successful</pre>
</details>
<details>
<summary>which -a php-fpm</summary>
<pre>/opt/homebrew/sbin/php-fpm
/usr/sbin/php-fpm</pre>
</details>
<details>
<summary>/opt/homebrew/opt/php/sbin/php-fpm -v</summary>
<pre>PHP 8.1.0 (fpm-fcgi) (built: Nov 29 2021 23:05:18)
Copyright (c) The PHP Group
Zend Engine v4.1.0, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.0, Copyright (c), by Zend Technologies</pre>
</details>
<details>
<summary>sudo /opt/homebrew/opt/php/sbin/php-fpm -y /opt/homebrew/etc/php/8.1/php-fpm.conf --test</summary>
<pre>[30-Nov-2021 01:11:55] NOTICE: configuration file /opt/homebrew/etc/php/8.1/php-fpm.conf test is successful</pre>
</details>
<details>
<summary>ls -al ~/Library/LaunchAgents | grep homebrew</summary>
<pre>-rw-r--r--   1 leon  staff   586 29 Nov 23:58 homebrew.mxcl.php.plist</pre>
</details>
<details>
<summary>ls -al /Library/LaunchAgents | grep homebrew</summary>
<pre></pre>
</details>
<details>
<summary>ls -al /Library/LaunchDaemons | grep homebrew</summary>
<pre>-rw-r--r--   1 root  admin   602 30 Nov 00:19 homebrew.mxcl.dnsmasq.plist
-rw-r--r--   1 root  admin   490 30 Nov 00:19 homebrew.mxcl.nginx.plist
-rw-r--r--   1 root  admin   586 30 Nov 00:19 homebrew.mxcl.php.plist</pre>
</details>
<details>
<summary>ls -al /Library/LaunchDaemons | grep "com.laravel.valet."</summary>
<pre></pre>
</details>
<details>
<summary>ls -aln /etc/resolv.conf</summary>
<pre>lrwxr-xr-x  1 0  0  22 10 Set 02:32 /etc/resolv.conf -> ../var/run/resolv.conf</pre>
</details>
<details>
<summary>cat /etc/resolv.conf</summary>
<pre>#
# macOS Notice
#
# This file is not consulted for DNS hostname resolution, address
# resolution, or the DNS query routing mechanism used by most
# processes on this system.
#
# To view the DNS configuration used by this system, use:
#   scutil --dns
#
# SEE ALSO
#   dns-sd(1), scutil(8)
#
# This file is automatically generated.
#
search station
nameserver 192.168.1.1</pre>
</details>
<details>
<summary>ifconfig lo0</summary>
<pre>lo0: flags=8049<UP,LOOPBACK,RUNNING,MULTICAST> mtu 16384
	options=1203<RXCSUM,TXCSUM,TXSTATUS,SW_TIMESTAMP>
	inet 127.0.0.1 netmask 0xff000000 
	inet6 ::1 prefixlen 128 
	inet6 fe80::1%lo0 prefixlen 64 scopeid 0x1 
	nd6 options=201<PERFORMNUD,DAD></pre>
</details>
<details>
<summary>sh -c 'echo "------\n/opt/homebrew/etc/nginx/valet/valet.conf\n---\n"; cat /opt/homebrew/etc/nginx/valet/valet.conf | grep -n "# valet loopback"; echo "\n------\n"'</summary>
<pre>------
/opt/homebrew/etc/nginx/valet/valet.conf
---

3:    #listen VALET_LOOPBACK:80; # valet loopback

------</pre>
</details>
<details>
<summary>sh -c 'for file in ~/.config/valet/dnsmasq.d/*; do echo "------\n~/.config/valet/dnsmasq.d/$(basename $file)\n---\n"; cat $file; echo "\n------\n"; done'</summary>
<pre>------
~/.config/valet/dnsmasq.d/tld-test.conf
---

address=/.test/127.0.0.1
listen-address=127.0.0.1

------</pre>
</details>
<details>
<summary>sh -c 'for file in ~/.config/valet/nginx/*; do echo "------\n~/.config/valet/nginx/$(basename $file)\n---\n"; cat $file | grep -n "# valet loopback"; echo "\n------\n"; done'</summary>
<pre>------
~/.config/valet/nginx/tailtest.test
---

3:    #listen 127.0.0.1:80; # valet loopback
10:    #listen 127.0.0.1:443 ssl http2; # valet loopback
54:    #listen 127.0.0.1:60; # valet loopback

------</pre>
</details>

**Comment by ugyballoons** [comment]
This looks old but just in case it's useful to anyone...
I had the same issue & assumed an incompatibility with PHP 8.1.1

Run `valet use php@8.0` twice
once for the install and again for valet to link it up

**Reply by settermjd** [reply]
I also had to run `composer global update`. After that, `valet share` worked properly.

**Comment by joostvanhoof** [comment]
@ugyballoons do you know how to use `valet share` with PHP8.1? Would like to use 8.1 for my projects but then `valet share` doesn't work...

**Reply by ugyballoons** [reply]
What errors are you getting?
I'm running Laravel Valet 2.18.6 which is using php 8.1 and `valet share` is working well...

**Reply by brainmaniac** [reply]
I need to upgrade my valet version:

```
composer global require laravel/valet --ignore-platform-reqs
```
